litgpt/finetune/full.py

```diff
@@ -150,6 +150,14 @@ def main(
     if fabric.device.type == "cuda":
         fabric.print(f"Memory used: {torch.cuda.max_memory_allocated() / 1e9:.02f} GB")
 
+    # Final evaluation
+    val_loss = validate(
+        fabric, model, val_dataloader, tokenizer, dataclasses.replace(eval, max_iters=len(val_dataloader)), data
+    )
+    metrics = {"val_loss": val_loss, "val_ppl": math.exp(val_loss)}
+    fabric.log_dict(metrics, step=state["iter_num"])
+    fabric.print(f"Final evaluation | val loss: {val_loss.item():.4f} | val ppl: {math.exp(val_loss):.4f}")
```

Comment:
I wonder what you think about computing this for the training dataset as well? I usually find it useful to look at validation and training performance side by side. It would be expensive though, so maybe a rolling average over the recent training set losses. Could be a separate PR.